rasterio/_io.pyx

```diff
@@ -1821,6 +1858,56 @@ cdef class DatasetWriterBase(DatasetReaderBase):
         """
         self.write(src, bidx, window=window)
 
+    def clear_stats(self):
+        """Clear stored statistics for all dataset bands.
+
+        Returns
+        -------
+        None
+        """
+        GDALDatasetClearStatistics(self._hds)
+
+    def update_stats(self, *, stats=None, indexes=None, approx=False):
```

Comment:
Just want to point out that these new methods have keyword-only arguments. So that you can't cross up `stats` and `indexes`.